src/citrine/resources/design_space.py

```diff
@@ -206,6 +227,54 @@ def create_default(self,
         data = self.session.post_resource(path, json=payload, version=self._api_version)
         return self.build(DesignSpace.wrap_instance(data["instance"]))
 
+    def convert_to_hierarchical(
+            self,
+            uid: Union[UUID, str],
+            *,
+            data_sources: Optional[List[DataSource]] = None,
+            template_link: Optional[TemplateLink] = None,
+            display_name: Optional[str] = None,
+    ) -> HierarchicalDesignSpace:
+        """Convert an existing ProductDesignSpace into an equivalent HierarchicalDesignSpace.
+
+        A :class:`~citrine.informatics.design_spaces.ProductDesignSpace` can be mapped to a
+        :class:`~citrine.informatics.design_spaces.HierarchicalDesignSpace` with a root node
+        containing the dimensions and formulation subspace of the original design space.
+        The resulting root node can be supplemented with the data sources, template link,
+        and display name provided to this method.
+
+        Data sources enable the Citrine Platform to design over "known" ingredients
+        found in the formulation subspace of the original design space.
+        These materials are looked up from the data source and injected into the material history
+        of the generated candidates.
+
+        Parameters
+        ----------
+        uid: Union[str, UUID]
+            UUID of the existing product design space to convert to a hierarchical version
+        data_sources: Optional[List[DataSource]]
+            Optional data sources to include in the converted hierarchical design space
+        template_link: Optional[TemplateLink]
+            Optional template link to include on the root material node
+        display_name: Optional[str]
+            Optional display name to include on the root material node
+
+        Returns
+        -------
+        A :class:`~citrine.informatics.design_spaces.HierarchicalDesignSpace`
+        with a single material node obtained from the input search space.
+
+        """
+        path = f"projects/{self.project_id}/design-spaces/{uid}/convert-hierarchical"
```

Comment:
The bound terms should really be run through percent encoding.

from citrine._utils.functions import format_escaped_url